.github/workflows/ci.yaml

```diff
@@ -65,3 +49,50 @@ jobs:
         with:
           test-results: test.json
 
+  generate-client:
+    runs-on: ubuntu-latest
+    env:
+      TESTNET_ACCOUNT: "98ebfd2fa655cea228bd307e0e838cf3bdf08f8dde0b9baa6054c54b462b3acc"
+      MAINNET_FIND: "98ebfd2fa655cea228bd307e0e838cf3bdf08f8dde0b9baa6054c54b462b3acc"
+      MAINNET_FIND_ADMIN: "98ebfd2fa655cea228bd307e0e838cf3bdf08f8dde0b9baa6054c54b462b3acc"
+    needs: test
+    output:
+      client_changed: ${{ steps.bump.outputs.bumped }}
+    steps:
+     - name: Checkout code
+       uses: actions/checkout@v3
+     - uses: actions/setup-go@v4
+       with:
+         go-version: 1.18
+     - run: make client
+     - name: verify client changed
+       uses: tj-actions/verify-changed-files@v14
+       id: verify-changed-files
+       with:
+         files: |
+           lib/find.json
+           dapper-tx
+      - name: fail if dapper transactions changed
+        if: contains(steps.verify-changed-files.outputs.changed_files, 'dapper-tx')
+        run: |
+        echo "dapper-tx is not updated"
+        exit 1
+      - name: fail if find client changed
+        if: contains(steps.verify-changed-files.outputs.changed_files, 'lib/find.json')
+        run: |
+        echo "find client is not updated"
+        exit 1
+
+      - name: Check package version
+        id: cpv
+        uses: PostHog/check-package-version@v2
+
+      - name: Ensure package is not rc version
+      if : contains(steps.cpv.outputs.committed-version, 'rc')
+        run : |
+          echo "version is not updated, it cannot be in rc version"
+          exit 1
+
+      - name: Publish new version
+        if: steps.cpv.outputs.is-new-version == 'true'
+        run: npm publish
```

Comment:
can we just run npm publish here? do we not need to set up permissions for it first and navigate to the correct folder?

Reply:
look in pr.yaml for the correct code here i assume.